src/features/comments.ts

```diff
@@ -0,0 +1,82 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Comment } from '../types/Comment';
+import { createComment, deleteComment, getPostComments } from '../api/comments';
+
+type CommentsState = {
+  items: Comment[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: CommentsState = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const loadComments = createAsyncThunk(
+  'comments/fetch',
+  async (postId: number) => {
+    const result = await getPostComments(postId);
+
+    return result;
+  },
+);
+
+export const addComment = createAsyncThunk(
+  'comments/post',
+  async (data: Omit<Comment, 'id'>) => {
+    const result = await createComment(data);
+
+    return result;
+  },
+);
+
+export const removeComment = createAsyncThunk(
+  'comments/delete',
+  async (commentId: number) => {
+    await deleteComment(commentId);
+
+    return commentId;
+  },
+);
+
+export const commentsSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder.addCase(loadComments.pending, state => {
+      return { ...state, loaded: false };
+    });
```

Comment:
In the `loadComments.pending` case, you're resetting the `loaded` state to `false`, but you're not resetting the `hasError` state. This could potentially lead to a situation where `hasError` is `true` and `loaded` is `false`, which might not be an expected state in your application. Consider resetting `hasError` to `false` here as well.